doteki/plugins/figlet.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import pyfiglet  # type: ignore
+from typing import Any
+
+
+def run(settings: dict[str, Any]) -> str | None:
+
+    if not validate_settings(settings):
+        return None
+    text = settings.get("ascii_text")
+    font = settings.get("font", "standard")
+    result = pyfiglet.figlet_format(text, font)
+    pre_text = "```text\n"
+    post_text = "\n```"
+    result = pre_text + result.rstrip() + post_text
+    return str(result)
+
+
+def validate_settings(settings: dict[str, Any]) -> bool:
+
+    text = settings.get("ascii_text")
+    font = settings.get("font", "standard")
+
+    errors = []
+
+    # Check required setting.
+    if "ascii_text" not in settings:
+        errors.append("No text provided for the FIGlet plugin")
+
+    try:
+        pyfiglet.figlet_format("test", font)
```

Comment:
Conceptually it makes sense to check this here, but it's not so clean in my opinion.

How about we move this try/except block into the `run` function? This way we can still catch and report the error, but avoid unnecessary `pyfiglet` calls.

Reply:
Okay, I think is a good idea. I'll do it.